src/in_progress.rs

```diff
@@ -1110,3 +1110,123 @@ pub fn is_hardware_waiting_repair(
+    let conn = get_connection_from_pool(pool)?;
+
+    // Get region Id from region name
+    let stmt = "SELECT region_id FROM regions WHERE region_name = $1";
+    let stmt_query = conn.query(stmt, &[&region_name])?;
+
+    if let Some(res) = stmt_query.into_iter().next() {
+        // Exists, return region_id
+        let id: i32 = res.get(0);
+        debug!("Region id {} for the region {}", id, region_name);
+        Ok(Some(id as u32))
+    } else {
+        // does not exist
+        debug!("No region with name {} in database", region_name);
+        Ok(None)
+    }
+    
+}
+
+/// Get storage id based on the storage type.
+pub fn get_storage_id(pool: &Pool<ConnectionManager>, storage_type: &str) -> BynarResult<Option<u32>> {
+    let conn = get_connection_from_pool(pool)?;
+
+    // Get storage Id from storage type
+    let stmt = "SELECT storage_id FROM storage_types WHERE storage_type= $1 ";
+    let stmt_query = conn.query(&stmt, &[&storage_type])?;
+
+    if let Some(res) = stmt_query.into_iter().next() {
+        // Exists, return storage_id
+        let id: i32 = res.get(0);
+        debug!(
+            "Storage id {} for the storage_type {}",
+            id, storage_type
+        );
+        Ok(Some(id as u32))
+    } else {
+        // does not exist
+        debug!("No storage with type {} in database", storage_type);
+        Ok(None)
+    }
+}
+
+/// Get storage id based on the storage type.
+pub fn get_storage_detail_id(
+    pool: &Pool<ConnectionManager>,
+    storage_id: u32,
+    region_id: u32,
+    host_name: &str,
+) -> BynarResult<Option<u32>> {
+    let conn = get_connection_from_pool(pool)?;
+
+    // Get storage detail Id
+    let stmt = "SELECT detail_id FROM storage_details WHERE storage_id = $1
+            AND region_id = $2 AND hostname = $3 ";
+    let stmt_query = conn.query(&stmt, &[&storage_id,&region_id, &host_name])?;
+
+    if let Some(res) = stmt_query.into_iter().next() {
+        // Exists, return storage_id
+        let id: i32 = res.get(0);
+        debug!(
+            "Storage details id {} for the host_name {} , region {} , storage_id {} ",
+            id, host_name, region_id, storage_id
+        );
+        Ok(Some(id as u32))
+    } else {
+        // does not exist
+        debug!(
+            "No storage detail id with host_name {} , region {} , storage_id {} 
+        in database",
+            host_name, region_id, storage_id,
+        );
+        Ok(None)
+    }
+}
+
+/// Get a list of ticket IDs (JIRA/other ids) that belong to all servers.
+/// that are in pending state  and outstanding tickets
+pub fn get_all_pendout_tickets(
+    pool: &Pool<ConnectionManager>
+) -> BynarResult<Vec<DiskRepairTicket>> {
+    let conn = get_connection_from_pool(pool)?;
+
+    // Get all tickets of myself with device.state=WaitingForReplacement and operation_detail.status = pending or in_progress
+     let stmt = "SELECT tracking_id, device_name, device_path FROM operation_details JOIN operations
```

Comment:
Should also include `storage_detail_id`. This should then be reverse mapped to retreive the hostname/server name. You may have to `DiskRepairTicket` to include an optional storage_detail_id or hostname or make another struct that `get_all_pending_tickets()` could return.